src/neurostatslib/glm.py

```diff
@@ -1,431 +1,598 @@
 
-        if init_spikes.shape[0] != bs.shape[0]:
+        n_basis_coupling = coupling_basis_matrix.shape[1]
+        n_neurons = self.baseline_link_fr_.shape[0]
+
+        if init_y is None:
+            init_y = jnp.zeros((coupling_basis_matrix.shape[0], n_neurons))
+
+        Wf = self.basis_coeff_[:, n_basis_coupling * n_neurons :]
+        Wr = self.basis_coeff_[:, : n_basis_coupling * n_neurons]
+        bs = self.baseline_link_fr_
+
+        feedforward_input, init_y = self.preprocess_simulate(
+            feedforward_input, params_f=(Wf, bs), init_y=init_y, params_r=(Wr, bs)
+        )
+
+        self._check_input_and_params_consistency(
+            (Wr, bs),
+            y=init_y,
+        )
+
+        if init_y.shape[0] != coupling_basis_matrix.shape[0]:
             raise ValueError(
-                "Number of neurons must be the same during simulation and fitting! "
-                f"init_spikes n_neurons: {init_spikes.shape[0]}, "
-                f"self.baseline_log_fr_ n_neurons: {self.baseline_log_fr_.shape[0]}"
+                "`init_y` and `coupling_basis_matrix`"
+                " should have the same window size! "
+                f"`init_y` window size: {init_y.shape[1]}, "
+                f"`coupling_basis_matrix` window size: {coupling_basis_matrix.shape[1]}"
             )
-        if init_spikes.shape[1] != self.spike_basis_matrix.shape[1]:
-            raise ValueError(
-                "init_spikes has the wrong number of time steps!"
-                f"init_spikes time steps: {init_spikes.shape[1]}, "
-                f"spike_basis_matrix window size: {self.spike_basis_matrix.shape[1]}"
+
+        subkeys = jax.random.split(random_key, num=feedforward_input.shape[0])
+        # (n_samples, n_neurons)
+        feed_forward_contrib = jnp.einsum("ik,tik->ti", Wf, feedforward_input)
+
+        def scan_fn(
+            data: Tuple[jnp.ndarray, int], key: jax.random.PRNGKeyArray
+        ) -> Tuple[Tuple[jnp.ndarray, int], Tuple[jnp.ndarray, jnp.ndarray]]:
+            """Scan over time steps and simulate activity and rates.
+
+            This function simulates the neural activity and firing rates for each time step
+            based on the previous activity, feedforward input, and model coefficients.
+            """
+            activity, chunk = data
+
+            # Convolve the neural activity with the coupling basis matrix
+            conv_act = convolve_1d_trials(coupling_basis_matrix, activity[None, :, :])[
+                0
+            ]
+
+            # Extract the corresponding slice of the feedforward input for the current time step
+            input_slice = jax.lax.dynamic_slice(
+                feed_forward_contrib,
+                (chunk, 0),
+                (1, feed_forward_contrib.shape[1]),
             )
 
-        subkeys = jax.random.split(random_key, num=n_timesteps)
+            # Reshape the convolved activity and concatenate with the input slice to form the model input
```

Comment:
reshape to what?

Reply:
I added a step by step description of the call